java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -75,11 +76,42 @@ public JObject(
     } else {
       String nextPackagePath = pkgPrefix + AbstractJSONSchema2Pojo.packageName(this.className);
 
+      // in order to handle duplicated fields, first let's build a map of fields grouped by their sanitized names, i.e.:
+      // 1(fieldName) -> n(fieldDefinition(key, props))
+      final Map<String, Map<String, JSONSchemaProps>> groupedFieldDefinitions = fields.entrySet().stream()
+          .collect(Collectors.groupingBy(
+              JObject::sanitizeFieldNameFromDefinition,
+              Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+      int deprecatedFieldCounter = 0;
+
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
+        String fieldKey = field.getKey();
+        // lookup the duplicated field properties map
+        final Map.Entry<String, Map<String, JSONSchemaProps>> fieldDuplicatesDefinition = groupedFieldDefinitions.entrySet()
+            .stream()
+            .filter(d -> d.getValue().containsKey(field.getKey()))
+            .findFirst().get();
+        if (fieldDuplicatesDefinition.getValue().entrySet().size() > 1) {
+          Set<Map.Entry<String, JSONSchemaProps>> fieldDuplicates = fieldDuplicatesDefinition.getValue().entrySet();
+          final Long duplicatesCount = fieldDuplicates.stream().count();
```

Comment:
length?

Reply:
Just avoid to convert to a stream :-)

Reply:
Indeed, makes sense :slightly_smiling_face: